src/extensions/endpoints/adoption-code/src/repositories/directus.ts

```diff
@@ -1,51 +1,63 @@
 import type { EndpointExtensionContext } from '@directus/extensions';
-import type { AdoptedProbe, Request } from '../index.js';
+import type { AdoptedProbe, ProbeToAdopt } from '../index.js';
 
-export const createAdoptedProbe = async (req: Request, probe: AdoptedProbe, context: EndpointExtensionContext) => {
-	const { services, database } = context;
+export const createAdoptedProbe = async (userId: string, probe: ProbeToAdopt, context: EndpointExtensionContext) => {
+	const { services, database, getSchema } = context;
 	const itemsService = new services.ItemsService('gp_probes', {
-		schema: req.schema,
+		schema: await getSchema({ database }),
+		knex: database,
 	});
 
-	const name = await getDefaultProbeName(req, probe, context);
+	const name = await getDefaultProbeName(userId, probe, context);
 
-	const adoption = {
+	const adoption: Omit<AdoptedProbe, 'id'> = {
 		ip: probe.ip,
+		altIps: probe.altIps,
 		name,
 		uuid: probe.uuid,
 		version: probe.version,
 		nodeVersion: probe.nodeVersion,
 		hardwareDevice: probe.hardwareDevice,
 		hardwareDeviceFirmware: probe.hardwareDeviceFirmware,
+		systemTags: probe.systemTags,
 		status: probe.status,
 		city: probe.city,
-		systemTags: probe.systemTags,
 		state: probe.state,
 		country: probe.country,
 		latitude: probe.latitude,
 		longitude: probe.longitude,
 		asn: probe.asn,
 		network: probe.network,
-		userId: req.accountability.user,
+		userId,
 		lastSyncDate: new Date(),
 		isIPv4Supported: probe.isIPv4Supported,
 		isIPv6Supported: probe.isIPv6Supported,
 	};
 
-	const existingProbe = await database('gp_probes').whereRaw(`
-		(
-			ip = ?
-			OR JSON_CONTAINS(altIps, ?)
-		)
-		AND userId IS NULL
-	`, [ probe.ip, `"${probe.ip}"` ]).first();
+	const existingProbe = await database('gp_probes')
+		.where({ uuid: probe.uuid })
+		.orWhere({ ip: probe.ip })
+		.orWhereRaw('JSON_CONTAINS(altIps, ?)', [ probe.ip ])
+		.first<AdoptedProbe>();
 
 	let id: string;
 
 	if (existingProbe) {
```

Comment:
It seems that if the user matches as well, we can return here? Then we could also make `adopt-by-token` `PUT` instead of `POST` and allow retries from the API side.